Match.cpp

```diff
@@ -331,21 +331,114 @@ void Match::print( ostream& fout )
 		fout.write( (char*) &hom[1] , sizeof( bool ) );
 	} else
 	{
-		fout << node[0]->getID() << '\t';
-		fout << node[1]->getID() << '\t';
-		fout << ALL_SNPS.getSNP(snp_start).getChr() << '\t';
-		fout << ALL_SNPS.getSNP(snp_start).getPhysPos() << ' ';
-		fout << ALL_SNPS.getSNP(snp_end).getPhysPos() << '\t';
-		fout << ALL_SNPS.getSNP(snp_start).getSNPID() << ' ';
-		fout << ALL_SNPS.getSNP(snp_end).getSNPID() << '\t';
-		fout << ( snp_end - snp_start + 1) << '\t';
-		fout << setiosflags(ios::fixed) << setprecision(2) << distance << '\t';
-		if ( genetic ) fout << "cM" << '\t'; else fout << "MB" << '\t';
-		fout << dif;
-		for ( int n = 0 ; n < 2 ; n++ )
-			if ( hom[n] ) fout << '\t' << 1; else fout << '\t' << 0;
-		fout << endl;
+		if ( ALL_SAMPLES.useEmbarkRFGermlineOutput ) {
+			int key1 = stoi(node[0]->single_id);
+			int key2 = stoi(node[1]->single_id);
+			vector<string> oline;
+			string joined_oline;
+			ofstream* ofs;
+			string chromosome = ALL_SNPS.getSNP(snp_start).getChr();
+			string start_pos = to_string(ALL_SNPS.getSNP(snp_start).getPhysPos());
+			string end_pos = to_string(ALL_SNPS.getSNP(snp_end).getPhysPos());
+			
+			// homoz
+			if ( key1 == key2 && node[0]->is_new ) {
+				// cout << "Writing out homoz" << endl;
+				oline.push_back(chromosome);
+				oline.push_back(start_pos);
+				oline.push_back(end_pos);
+				ofs = node[0]->getIndividualHomozFile();
+				join(oline, '\t', joined_oline);
+				*ofs << joined_oline << endl;
+			}
+			// key1 is new, key2 is old
+			else if ( key1 > key2 && node[0]->is_new ) {
+				// cout << "Writing match tracts for new key1 " << key1 << endl;
+				oline.push_back(node[0]->single_id);
+				oline.push_back(node[0]->haplotype);
+				oline.push_back(node[1]->single_id);
+				oline.push_back(node[1]->haplotype);
+				oline.push_back(chromosome);
+				oline.push_back(start_pos);
+				oline.push_back(end_pos);
+				ofs = node[0]->getIndividualMatchFile();
+				join(oline, '\t', joined_oline);
+				*ofs << joined_oline << endl;
+			}
+			// key2 is new, key1 is old
+			else if ( key2 > key1 && node[1]->is_new ) {
+				// cout << "Writing match tracts for new key2 " << key2 << endl;
+				oline.push_back(node[1]->single_id);
+				oline.push_back(node[1]->haplotype);
+				oline.push_back(node[0]->single_id);
+				oline.push_back(node[0]->haplotype);
+				oline.push_back(chromosome);
+				oline.push_back(start_pos);
+				oline.push_back(end_pos);
+				ofs = node[1]->getIndividualMatchFile();
+				join(oline, '\t', joined_oline);
+				*ofs << joined_oline << endl;
+			}
+			// newdog : newdog comparison, write out same record twice
+			else if ( node[0]->is_new  &&  node[1]->is_new  && key1 != key2 ) {
```

Comment:
QQ: Is this block reachable? I'm a little hazy on what `node[*]` is. For instance, in the line above:
```
key1 > key2 && node[0]->is_new
```
... Does this necessarily imply that only one of the keys is `new`?

I'm naively assuming `node[0]->is_new` asserts that the first key is `new`, but doesn't say anything about the second...

Reply:
Correct, so this specific case is the `newdog` `newdog` comparison, where both keys are new and not the same. This would require us to output the match twice, one for each new dog. This block is definitely reachable only when we run the compset `9999`.

> key1 > key2 && node[0]->is_new
> ```
> 
> ... Does this necessarily imply that only one of the keys is `new`?
> 
> I'm naively assuming `node[0]->is_new` asserts that the first key is `new`, but doesn't say anything about the second...

Correct, this is checking only if the first key is new. The per-dog matches file require(?) that the new dog be first, but germline doesn't care, so each line needs to be checked to see which key is the new dog and put that one first. This logic is lifted from the rust script

Reply:
Again, this only matters when we are comparing a set of old dogs against new dogs, and using the `-samples-to-compare` and `-new-samples` flags. Otherwise, germline will do all dogs against all dogs in the ped file by default. This doesn't take advantage of the fact that new keys will be greater than old keys as I prefer to be more explicit about checking if a key is new or not.